tech_soln/real_time_dashboards.rst

```diff
@@ -0,0 +1,422 @@
+Citus on either a single or multiple machines you're ready to try it out.
+
+Data Model
+----------
+
+The data we're dealing with is an immutable stream of log data. We'll insert directly into
+Citus but it's also common for this data to first be routed through something like Kafka.
+Doing so has the usual advantages, and makes it easier to pre-aggregate the data once data
+volumes become unmanageably high.
+
+We'll use a simple schema for ingesting HTTP event data. This schema serves as an example
+to demonstrate the overall architecture; a real system might use additional columns.
+
+.. code-block:: sql
+
+  -- this is run on the master
+  CREATE TABLE http_request (
+    site_id INT,
+    ingest_time TIMESTAMPTZ DEFAULT now(),
+
+    url TEXT,
+    request_country TEXT,
+    ip_address INET,
+
+    status_code INT,
+    response_time_msec INT
+  );
+  SELECT master_create_distributed_table('http_request', 'site_id', 'hash');
+  SELECT master_create_worker_shards('http_request', 16, 2);
+
+When we call :ref:`master_create_distributed_table <master_create_distributed_table>`
+we ask Citus to hash-distribute ``http_request`` using the ``site_id`` column. That means
+all the data for a particular site will live in the same shard.
+
+When we call :ref:`master_create_worker_shards <master_create_worker_shards>` we tell
+Citus to create 16 shards and 2 replicas of each shard (for a total of 32 shard replicas).
+We recommend :ref:`using 2-4x as many shards <faq_choose_shard_count>` as CPU cores in
+your cluster. Using this many shards lets you rebalance data across your cluster after
+adding new worker nodes.
+
+Using a replication factor of 2 means every shard is held on multiple workers. When a
+worker fails the master will prevent downtime by serving queries for that worker's shards
+using the other replicas.
+
+.. NOTE::
+
+  In Citus Cloud you must use a replication factor of 1 (instead of the 2 used here). As
+  Citus Cloud uses `streaming replication
+  <https://www.postgresql.org/docs/current/static/warm-standby.html>`_ to achieve high
+  availability maintaining shard replicas would be redundant.
+
+With this, the system is ready to accept data and serve queries! We've provided `a data
+ingest script
+<https://github.com/citusdata/realtime-dashboards-resources/blob/master/ingest_example_data.sql>`_
+you can run to generate example data. Once you've ingested data, you can run dashboard
+queries such as:
+
+.. code-block:: sql
+
+  SELECT
+    date_trunc('minute', ingest_time) as minute,
+    COUNT(1) AS request_count,
+    SUM(CASE WHEN (status_code between 200 and 299) THEN 1 ELSE 0 END) as success_count,
+    SUM(CASE WHEN (status_code between 200 and 299) THEN 0 ELSE 1 END) as error_count,
+    SUM(response_time_msec) / COUNT(1) AS average_response_time_msec
+  FROM http_request
+  WHERE zone_id = 1 AND minute > date_trunc('minute', now()) - interval '5 minutes'
```

Comment:
I think it is not possible to use ALIAS in WHERE clause. It gives me error; 
ERROR:  column "minute" does not exist

Related: https://www.postgresql.org/message-id/18195.1038530280@sss.pgh.pa.us
I am not sure if this is still the case, but probably it is.